test/src/unittests/tonal/test_pitchsaliencefunction.py

```diff
@@ -0,0 +1,223 @@
+# the terms of the GNU Affero General Public License as published by the Free
+# Software Foundation (FSF), either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the Affero GNU General Public License
+# version 3 along with this program. If not, see http://www.gnu.org/licenses/
+
+
+from numpy import *
+from essentia_test import *
+
+class TestPitchSalienceFunction(TestCase):
+  
+    def testInvalidParam(self):
+        self.assertConfigureFails(PitchSalienceFunction(), {'binResolution': -1})
+        self.assertConfigureFails(PitchSalienceFunction(), {'harmonicWeight': -1})
+        self.assertConfigureFails(PitchSalienceFunction(), {'harmonicWeight': 2})
+        self.assertConfigureFails(PitchSalienceFunction(), {'magnitudeCompression': -1})
+        self.assertConfigureFails(PitchSalienceFunction(), {'magnitudeCompression': 2})        
+        self.assertConfigureFails(PitchSalienceFunction(), {'magnitudeThreshold': -1})
+        self.assertConfigureFails(PitchSalienceFunction(), {'numberHarmonics': 0})
+        self.assertConfigureFails(PitchSalienceFunction(), {'numberHarmonics': -1})        
+        self.assertConfigureFails(PitchSalienceFunction(), {'referenceFrequency': -1})
+
+    def testEmpty(self): 
+        self.assertEqualVector(PitchSalienceFunction()([], []), zeros(600))
+
+    # Provide a single input peak with a unit magnitude at the reference frequency, and 
+    # validate that the output salience function has only one non-zero element at the first bin.
+    def testSinglePeak(self):
+        freq_speaks = [55] # length 1
+        mag_speaks = [1] # length 4
+        calculatedPitchSalience = PitchSalienceFunction()(freq_speaks,mag_speaks)    
+        self.assertEqual(calculatedPitchSalience[0],1)
+
+    def test3Peaks(self):
+        freq_speaks = [55, 100, 340] # length 1
+        mag_speaks = [1, 1, 1] # length 4
+        # For a single frequency 55 Hz with unitary  amplitude the 10 non zero salience function values are the following
+        calculatedPitchSalience = PitchSalienceFunction()(freq_speaks,mag_speaks)    
+        self.assertAlmostEqual(calculatedPitchSalience[0], 1.16384, 3)
+        self.assertEqual(len(calculatedPitchSalience), 600)        
+
+    def test3PeaksHw0(self):
+        freq_speaks = [55, 100, 340] # length 1
+        mag_speaks = [1, 1, 1] # length 4
+        # For a single frequency 55 Hz with unitary  amplitude the 10 non zero salience function values are the following
+
+        calculatedPitchSalience = PitchSalienceFunction(harmonicWeight=0)(freq_speaks, mag_speaks)    
+        self.assertAlmostEqual(calculatedPitchSalience[0], 1.16384, 3)
```

Comment:
Just to clarify the algorithm: Here if `harmonicWeight = 0` should we expect the output value at the 0th bin to be 1?